### ISSUE TITLE
Minor issue: unnatural classifier for FashionMNIST in 18.1.3

### ISSUE THREAD

**Issue by particle1331** [d2l-ai/d2l-en#1632]
The classifier used in the text seems rather unnatural. A more natural way is to flatten the images, normalize them, then take the dot product as discussed in the text as a measure of similarity. The predicted label is the label of the average image that is more similar to the test image, hence the argmax.

```
# normalize matrices by broadcasting
W = torch.stack([ave_0.flatten().t(), ave_1.flatten().t()], dim=1)
W = W / torch.norm(W, dim=0).reshape(1, -1)
X_test = X_test.reshape(-1, 784)
X_test = X_test / torch.norm(X_test, dim=1).reshape(-1, 1)

# predict and evaluate
y_pred = torch.argmax(X_test @ W, dim=1)
print((y_test == y_pred).type(torch.float).mean())
```

This obtains an accuracy of ~0.95. In contrast, the classifier used in the text is 

```
W = torch.stack([ave_0.flatten() - ave_1.flatten().t()], dim=1)
y_pred = (X_test @ W > - 1500000).type(torch.int)
```

One problem is that the broadcasting part for normalizing might be confusing at first reading. But I think this is good practice after reading section 2.2. I can open a PR for the three frameworks if you think this is worth changing. :-)
